pkg/agent/agent_windows.go

```diff
@@ -32,3 +40,127 @@ func (i *Initializer) setupExternalConnectivity() error {
+	i.nodeConfig.UplinkNetConfig.DNSServers = dnsServers
+	// Create HNS network.
+	return util.PrepareHNSNetwork(i.nodeConfig.PodCIDR, i.nodeConfig.NodeIPAddr, adapter)
+}
+
+// cleanupHostNetwork deletes HNS Network created by Antrea.
+func (i *Initializer) cleanupHostNetwork() error {
+	return util.DeleteHNSNetwork(util.LocalHNSNetwork)
+}
+
+// prepareOVSBridge adds local port and uplink to ovs bridge.
+func (i *Initializer) prepareOVSBridge() error {
+	hnsNetwork, err := hcsshim.GetHNSNetworkByName(util.LocalHNSNetwork)
+	if err != nil {
+		return err
+	}
+
+	// Set datapathID of OVS bridge.
+	// If no datapathID configured explicitly, the reconfiguration operation will change OVS bridge datapathID
+	// and break the OpenFlow channel.
+	// The length of datapathID is 64 bits, the lower 48-bits are for a MAC address, while the upper 16-bits are
+	// implementer-defined. Antrea uses "0x0000" for the upper 16-bits.
+	datapathID := strings.Replace(hnsNetwork.SourceMac, ":", "", -1)
+	datapathID = "0000" + datapathID
+	if err = i.ovsBridgeClient.SetDatapathID(datapathID); err != nil {
+		klog.Errorf("Failed to set datapath_id %s: %v", datapathID, err)
+		return err
+	}
+
+	// Create local port.
+	brName := i.ovsBridgeClient.GetBridgeName()
+	if _, err := i.ovsBridgeClient.GetOFPort(brName); err == nil {
+		klog.Infof("OVS bridge local port %s already exists, skip the configuration", brName)
+	} else {
+		if _, err := i.ovsBridgeClient.CreateInternalPort(brName, config.AutoAssignedOFPort, nil); err != nil {
+			return err
+		}
+	}
+
+	// If uplink is already exists, return.
+	uplinkNetConfig := i.nodeConfig.UplinkNetConfig
+	uplink := uplinkNetConfig.Name
+	if uplink == "" {
+		return fmt.Errorf("uplink network configuration not found")
+	}
+	if _, err := i.ovsBridgeClient.GetOFPort(uplink); err == nil {
+		klog.Infof("Uplink %s already exists, skip the configuration", uplink)
+		return err
+	}
+	// Create uplink port.
+	uplinkPortUUId, err := i.ovsBridgeClient.CreateUplinkPort(uplink, config.UplinkOFPort, nil)
+	if err != nil {
+		klog.Errorf("Failed to add uplink port %s: %v", uplink, err)
+		return err
+	}
+	uplinkInterface := interfacestore.NewUplinkInterface(uplink)
+	uplinkInterface.OVSPortConfig = &interfacestore.OVSPortConfig{uplinkPortUUId, config.UplinkOFPort}
+	i.ifaceStore.AddInterface(uplinkInterface)
+
+	// Move network configuration of uplink interface to OVS bridge local interface.
+	// - The net configuration of uplink will be restored by OS if the attached HNS network is deleted.
+	// - When ovs-switchd is down, antrea-agent will disable OVS Extension. The OVS bridge local interface will work
+	//   like a normal interface on host and is responsible for forwarding host traffic.
+	err = util.EnableHostInterface(brName)
+	if err = util.SetAdapterMACAddress(brName, &uplinkNetConfig.MAC); err != nil {
+		return err
+	}
+	// Remove the IP addresses if exists to avoid a candidate error of "Instance MSFT_NetIPAddress already exists" when
```

Comment:
Remove the existing IP address to avoid...

Reply:
Done